specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+
+If one of the entries inserted by the end developer is selected, the CustomMenuItemSelected event will be raised on the context menu item object that was selected.
+
+# Examples
+
+## Win32 C++ Add or Remove Entries From WebView Menu
+
+The developer can add or remove entries to the default WebView context menu. For this case, the developer specifies Handled to be false and is able to add or remove items from the collection of context menu items.
+
+ ```cpp
+    webview2->add_ContextMenuRequested(
+        Callback<ICoreWebView2ContextMenuRequestedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2ContextMenuRequestedEventArgs* args)
+            {
+                wil::com_ptr<ICoreWebView2ContextMenuItemCollection> items;
+                CHECK_FAILURE(args->get_MenuItems(&items));
+                wil::com_ptr<ICoreWebView2ContextMenuInfo> info;
+                CHECK_FAILURE(args->get_ContextMenuInfo(&info));
+                COREWEBVIEW2_CONTEXT_KIND context;
+                CHECK_FAILURE(info->get_ContextKind(&context));
+                CHECK_FAILURE(args->put_Handled(false));
+                UINT32 itemsCount;
+                CHECK_FAILURE(items->get_Count(&itemsCount));
+                // Removing the 'Save image as' context menu item for image context selections.
+                if (context == COREWEBVIEW2_CONTEXT_KIND_IMAGE)
+                {
+                    UINT32 removeIndex = itemsCount;
+                    wil::com_ptr<ICoreWebView2ContextMenuItem> current;
+                    for(UINT32 i = 0; i < itemsCount; i++) 
+                    {
+                        CHECK_FAILURE(items->GetValueAtIndex(i, &current));
+                        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR desc;
+                        CHECK_FAILURE(current->get_Descriptor(&desc));
+                        if (desc == COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_IMAGE_AS)
+                        {
+                            removeIndex = i;
+                        }
+                    }
+                    if (removeIndex < itemsCount)
+                    {
+                        CHECK_FAILURE(items->RemoveValueAtIndex(removeIndex));
+                    }
+                }
+                // Adding a custom context menu item for the page that will display the page's URI.
+                else if (context == COREWEBVIEW2_CONTEXT_KIND_PAGE)
+                {
+                    wil::com_ptr<ICoreWebView2Environment> webviewEnvironment;
+                    CHECK_FAILURE(m_appWindow->GetWebViewEnvironment()->QueryInterface(
+                        IID_PPV_ARGS(&webviewEnvironment)));
+                    wil::com_ptr<ICoreWebView2ContextMenuItem> newMenuItem;
+                    CHECK_FAILURE(webviewEnvironment->CreateContextMenuItem(L"Display page Uri", L"Shortcut", nullptr, COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_NORMAL, true, false, &newMenuItem));
```

Comment:
The two Boolean parameters at the end are basically meaningless when reading. Perhaps they should be combined into a `COREWEBVIEW2_CONTEXT_MENU_ITEM_FLAGS`?

```cpp
typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_FLAGS
{
    COREWEBVIEW2_CONTEXT_MENU_ITEM_FLAGS_NONE,
    COREWEBVIEW2_CONTEXT_MENU_ITEM_FLAGS_ENABLED,
    COREWEBVIEW2_CONTEXT_MENU_ITEM_FLAGS_CHECKED,
} COREWEBVIEW2_CONTEXT_MENU_ITEM_FLAGS;
```
Then the line becomes
```cpp
CHECK_FAILURE(webviewEnvironment->CreateContextMenuItem(L"Display page Uri", L"Shortcut", nullptr, COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_NORMAL, COREWEBVIEW2_CONTEXT_MENU_ITEM_FLAGS_ENABLED, &newMenuItem));
```